backend/curriculum_tracking/api_views.py

```diff
@@ -883,6 +891,26 @@ def auto_assign_reviewers(self, request, pk=None):
             response = actor.send()
             return Response({"status": "OK", "data": response.asdict()})
 
+    @action(
+        detail=False,
+        methods=["POST"],
+        serializer_class=serializers.SetDueTimeSerializer,
+        permission_classes=[HasObjectPermission(permissions=Team.PERMISSION_MANAGE_CARDS)]
+    )
+    def bulk_set_due_dates(self, request, pk=None):
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            teams = self.get_object()   # This will get you all the teams, we only need the teams in the request data
+            for team in teams:
+                if team.name in request.data.get('team'):
+                    team_cards = get_team_cards(team, request.data.get('content_item'))
+
+            for card in team_cards:
+                AgileCard.set_due_time(card, request.data.get('due_time'))
```

Comment:
Please use oop in conventional ways: `card.set_due_time(request.data.get('due_time'))`

`SomeClass.any_instance_method` is always bad practice.
`SomeClass.a_class_method` is good practice
`SomeClass.a_static_method` is also good